packages/ckeditor5-engine/src/model/schema.js

```diff
@@ -219,15 +219,28 @@ export default class Schema {
 	}
 
 	/**
-	 * Returns `true` if the given item is defined to be
-	 * a limit element by {@link module:engine/model/schema~SchemaItemDefinition}'s `isLimit` or `isObject` property
-	 * (all objects are also limits).
+	 * Returns `true` if the given item is should be treated as a limit element.
+	 * If it should act as a boundary between actions inside and outside.
+	 * It considers the item to be a limit element if its
+	 * {@link module:engine/model/schema~SchemaItemDefinition}'s
+	 * {@link module:engine/model/schema~SchemaItemDefinition#isLimit `isLimit`} or
+	 * {@link module:engine/model/schema~SchemaItemDefinition#isObject `isObject`} property
+	 * were set to `true`.
 	 *
 	 *		schema.isLimit( 'paragraph' ); // -> false
 	 *		schema.isLimit( '$root' ); // -> true
 	 *		schema.isLimit( editor.model.document.getRoot() ); // -> true
 	 *		schema.isLimit( 'image' ); // -> true
 	 *
+	 * See the {@glink framework/guides/deep-dive/schema "Schema" deep dive} guide.
+	 *
+	 * It considers `isObject` with higher priority than `isLimit`. Meaning, even if
+	 * `isLimit` was explicitly  set to `false` in `SchemaItemDefinition`, but `isObject` was set to `true`,
+	 * `isLimit()` will return `true`.
+	 *
+	 * **Note**: In the above scenario {@link module:engine/model/schema~Schema#getDefinition `schema.getDefinition( item ).isLimit`}
+	 * will be `false`, to match the exact definition given.
```

Comment:
As I mentioned before, I think this clarification about `Schema#getDefinition()` is unnecessary. 

> It considers \`isObject\` with higher priority than \`isLimit\`

Also, these API docs explain the implementation (which is very simple and anyone can see that) instead of the meaning (which is unclear when looking at the code). What we only need to know is:

*   that every object is a limit,
*   that a limit is a boundary that model selection cannot cross (both from the inside or from the outside),
*   that there's no operation in the editor that would end up in a the limit split into more fragments.

Reply:
Another thing that is missing is a mention that every object is a limit but not very limit is an object.